Client/Frontend/UserContent/UserScripts/MainFrame/AtDocumentStart/TrackingProtectionStats.js

```diff
@@ -0,0 +1,136 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+(function() {
+  "use strict";
+
+  Object.defineProperty(window.__firefox__, "TrackingProtectionStats", {
+    enumerable: false,
+    configurable: false,
+    writable: false,
+    value: { enabled: false }
+  });
+
+  Object.defineProperty(window.__firefox__.TrackingProtectionStats, "setEnabled", {
+    enumerable: false,
+    configurable: false,
+    writable: false,
+    value: function(enabled, securityToken) {
+      if (securityToken !== SECURITY_TOKEN) {
+        return;
+      }
+    
+      if (enabled === window.__firefox__.TrackingProtectionStats.enabled) {
+        return;
+      }
+
+      window.__firefox__.TrackingProtectionStats.enabled = enabled;
+
+      injectStatsTracking(enabled);
+    }
+  })
+
+  function onLoadNativeCallback() {
+    var messageHandler = window.webkit.messageHandlers.focusTrackingProtection;
+    var sendMessage = function(url) { messageHandler.postMessage({ url: url }) };
+
+    // Send back the sources of every script and image in the dom back to the host applicaiton
+    Array.prototype.map.call(document.scripts, function(t) { return t.src }).forEach(sendMessage);
+    Array.prototype.map.call(document.images, function(t) { return t.src }).forEach(sendMessage);
+  }
+
+  let originalOpen = null;
+  let originalSend = null;
+  let originalImageSrc = null;  
+  let mutationObserver = null;
+
+  function injectStatsTracking(enabled) {
+    if (enabled) {
```

Comment:
Changes from focus: I added the enable/disable block, and moved the postload js injection into a window.addEventListener(load)